installer/installer.go

```diff
@@ -208,6 +208,22 @@ func (i *Installer) GetArtifactName() string {
 	return i.ar.GetArtifactName()
 }
 
+func (i *Installer) GetArtifactProvides() *artifact.ArtifactProvides {
+	var provides *artifact.ArtifactProvides
+	if i.ar.GetInfo().Version >= 3 {
```

Comment:
I think you don't need this check, `Reader` already does this check for you (it's an interface). And the fewer places we need to make version comparisons, the better! :-)

Same for `GetArtifactDepends`.